dashboard/app/controllers/browser_events_controller.rb

```diff
@@ -0,0 +1,69 @@
+# A controller for reporting browser logs and metrics to Cloudwatch
+
+class BrowserEventsController < ApplicationController
+  EXPERIMENT_FLAG_NAME = 'browser-cloudwatch-metrics'
+  FIREHOSE_STUDY_NAME = 'browser-cloudwatch-metrics-errors'
+
+  LOGS_CLIENT = Aws::CloudWatchLogs::Client.new
+  LOG_GROUP_NAME_PREFIX = 'browser-event-logs-'
+
+  before_action :check_preconditions
+
+  # POST /put_logs
+  def put_logs
+    body = JSON.parse(request.body.read)
+
+    return render status :bad_request, json: {message: 'missing required params: logs'} unless body["logs"]
+
+    logs = body["logs"].map do |log_payload|
+      {timestamp: (Time.now.to_f * 1000).to_i, message: decorate_log_payload(log_payload)}
+    end
+
+    resp = LOGS_CLIENT.put_log_events(
+      {
+        log_group_name: "#{LOG_GROUP_NAME_PREFIX}#{Rails.env}",
+        log_stream_name: Rails.env.to_s,
+        log_events: logs
```

Comment:
I believe these need to be sorted by timestamp, worth doing that here or after the `map`?
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-cloudwatch-logs/classes/putlogeventscommand.html

Reply:
Well so since we're assigning timestamps here in the controller, all events should have the same timestamp. (that does beg the question - should we even allow this controller action to accept multiple log events? I kept it like this for flexibility in case we find sending multiple events at once to be useful)

Reply:
In theory the map should also keep them in order even if we change milliseconds while iterating over the logs. So we're probably fine.